pyanaconda/anaconda.py

```diff
@@ -165,9 +165,19 @@ def storage(self):
         if not self._storage:
             import blivet
             self._storage = blivet.Blivet(ksdata=self.ksdata)
-
-            if self.instClass.defaultFS:
-                self._storage.setDefaultFSType(self.instClass.defaultFS)
+            fstype = None
+
+            # Get the default fstype from a kickstart file.
+            if self.ksdata.autopart.autopart and self.ksdata.autopart.fstype:
+                fstype = self.ksdata.autopart.fstype
+            # Or from an install class.
+            elif self.instClass.defaultFS:
+                fstype = self.instClass.defaultFS
+
+            # Set the default fstype.
+            if fstype:
+                self._storage.setDefaultFSType(fstype)
+                self._storage.setDefaultBootFSType(fstype)
```

Comment:
We already check the value of `fstype` in `AutoPart.parse`. Also do really we want to replace the message from blivet with another one? Blivet raises `ValueError("unrecognized value %s for new default fs type" % newtype)` or `ValueError("new value %s is not valid as a default fs type" % fmt)`.